### PR TITLE
Add slash command access to make unit

### PR DESCRIPTION
**Pull request by bentito**
**Description of the change:**
This should allow PR comments of the form:
`/retest unit` to trigger `make unit`

**Motivation for the change:**
Prow offers PR comments to trigger fine-grained retest capabilities for tests. This trials adding this capability for the unit test target.

**Architectural changes:**

None

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

File: .github/workflows/unit.yml [hunk 1]
```diff
@@ -8,8 +8,13 @@ on:
   pull_request:
   workflow_dispatch:
   merge_group:
+  issue_comment:
+    types: [created]
 jobs:
   unit:
+    if: >-
+      github.event_name == 'pull_request' || 
+      (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/retest unit'))
```
[the remark this answers is not in the document]
Comment (bentito) [reply .github/workflows/unit.yml:17]: `contains` might trigger on comments unintentionally, startsWith forces to a smaller trigger.

This syntax is meant to be like Prow commands which many users of Prow based workflow are used to.

Was hoping to just start with a simple one here and we can add other commands as time and interest allows.

**Review by kevinrizza — DISMISSED** [review]

File: .github/workflows/unit.yml (hunk 1 above)
Comment (m1kola) [inline .github/workflows/unit.yml:17]: I think this will mean we do not run unit tests on merge queue, etc.

**Review by m1kola — CHANGES_REQUESTED** [review]
I think this change makes merge queue and other events to skip unit tests.

File: .github/workflows/unit.yml (hunk 1 above)
Comment (bentito) [reply .github/workflows/unit.yml:17]: in the `on` section we have `on:
on-push-branches:- master`  that should trigger the workflow on commits pushed to master, including merges. But just to be sure, and more clear, we add other events and comments to the `on` section in f6b3a1e4

Comment (m1kola) [reply .github/workflows/unit.yml:17]: I think `on` section was fine as it was originally. I would suggest reverting back.

My concern is within `if` which makes the `unit` job conditional. As it is right now - I think the workflow will be triggered on multiple events (push to `master`, pull request, merge queue, comment, etc), but the `unit` job will only be run on pull request events and comment events).

Or am I missing something?

Comment (bentito) [reply .github/workflows/unit.yml:17]: Ah, I think I get the problem you're finding. Yeah I think `on` can get reverted but `if` section needs to get modded to allow in all the other events we want unit to be run:
```yaml
jobs:
  unit:
    if: >-
      github.event_name == 'push' ||
      github.event_name == 'pull_request' ||
      github.event_name == 'workflow_dispatch' ||
      github.event_name == 'merge_group' ||
      (github.event_name == 'issue_comment' && startsWith(github.event.comment.body, '/retest unit'))
```
Seems right?

Comment (m1kola) [reply .github/workflows/unit.yml:17]: Maybe something like `github.event_name != 'issue_comment' || startsWith(github.event.comment.body, '/retest unit')`?

So we don't have to update in two places if we change something in `on`?

Comment (bentito) [reply .github/workflows/unit.yml]: Yep, that's better! 39d866f cleans up the on section and adds that to if section

**Review by m1kola — APPROVED** [review]
Looks good to me. I'm approving, but if you could squash into one commit - that would be great.

**Comment by bentito** [comment]
e2e (3) seems to fail on a "Subscription when unpacking bundle with bundle which OLM will fail to unpack [It] should report deprecated status when catalog is updated to deprecate an installed bundle"?

Seems unrelated to this PR. Merging again as first check on whether this is a flaky e2e test or something that needs fixing to unblock merges.

**Pull request merged**